File: app/src/main/java/com/motrixi/datacollection/fragment/PrivacyStatementFragment.java
package com.motrixi.datacollection.fragment;

import android.graphics.Color;
import android.os.Bundle;
import android.provider.Settings;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.text.Html;
import android.text.Spannable;
import android.text.SpannableStringBuilder;
import android.text.method.LinkMovementMethod;
import android.text.style.ClickableSpan;
import android.text.style.URLSpan;
import android.util.Log;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.ScrollView;
import android.widget.TextView;
import android.widget.Toast;

import com.motrixi.datacollection.DataCollectionActivity;
import com.motrixi.datacollection.content.Contants;
import com.motrixi.datacollection.content.Session;
import com.motrixi.datacollection.network.PostMethodUtils;
import com.motrixi.datacollection.utils.CustomStyle;
import com.motrixi.datacollection.utils.DisplayUtil;

import java.util.HashMap;

import static android.view.Gravity.CENTER;
import static android.widget.RelativeLayout.ALIGN_PARENT_BOTTOM;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link PrivacyStatementFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class PrivacyStatementFragment extends Fragment {

    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;
    private View rootView;
    private static DataCollectionActivity parentActivity;
    private RelativeLayout privateLayout;
    private LinearLayout actionBarLayout;
    private TextView tvTitle;
    private TextView tvContent1;
    private TextView tvCancel;
    private TextView tvOption;
    private TextView tvConfirm;

    public PrivacyStatementFragment() {
        // Required empty public constructor

        parentActivity = DataCollectionActivity.getSharedMainActivity();
        /*if (Contants.mFREContext != null) {
            mSession = new Session(Contants.mFREContext);
        } else {
            mSession = new Session(parentActivity);
        }*/
    }

    public static PrivacyStatementFragment newInstance(String param1, String param2) {
        parentActivity = DataCollectionActivity.getSharedMainActivity();

        PrivacyStatementFragment fragment = new PrivacyStatementFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }

        //mSession = new Session(Contants.mFREContext);
        //parentActivity = DataCollectionActivity.getSharedMainActivity();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        //return inflater.inflate(R.layout.fragment_privacy_statement2, container, false);

        initLayout();
        rootView = privateLayout;

        return rootView;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        //parentActivity = (DataCollectionActivity) Contants.mFREContext.getActivity();
        //mSession = new Session(Contants.mFREContext);

        customActionBarView();

        //Log.d("fragment", mSession.getConsentDataInfo().toString());
        setData();
    }

    private void initLayout() {
        privateLayout = new  RelativeLayout(getActivity());
        RelativeLayout.LayoutParams rootParams = new RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
        );

        RelativeLayout topLayout = new RelativeLayout(getActivity());
        RelativeLayout.LayoutParams topParams = new RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                DisplayUtil.dp2px(getActivity(), 50)
        );
        topLayout.setId(Contants.PRIVACY_TOP_ID);
        topLayout.setLayoutParams(topParams);
        topLayout.setBackgroundColor(Color.rgb(0, 150, 182));
        privateLayout.addView(topLayout);

        tvTitle = new TextView(getActivity());
        RelativeLayout.LayoutParams titleParams = new  RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.WRAP_CONTENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
        );
        titleParams.addRule(RelativeLayout.CENTER_IN_PARENT);
        //tvTitle.text = "Consent"
        tvTitle.setTextSize(20F);
        tvTitle.setTextColor(Color.BLACK);
        tvTitle.setLayoutParams(titleParams);
        topLayout.addView(tvTitle);


        ScrollView scrollView= new ScrollView(getActivity());
        RelativeLayout.LayoutParams scrollParams = new RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
        );
        scrollParams.bottomMargin = DisplayUtil.dp2px(getActivity(), 70);
        scrollParams.addRule(RelativeLayout.BELOW, topLayout.getId());
        scrollView.setLayoutParams(scrollParams);
        privateLayout.addView(scrollView);

        LinearLayout contentLayout = new LinearLayout(getActivity());
        contentLayout.setOrientation(LinearLayout.VERTICAL);
        RelativeLayout.LayoutParams contentParams = new RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
        );
        contentParams.setMargins(DisplayUtil.dp2px(getActivity(), 15),DisplayUtil.dp2px(getActivity(), 15),
                DisplayUtil.dp2px(getActivity(), 15),DisplayUtil.dp2px(getActivity(), 15));
        contentLayout.setLayoutParams(contentParams);
        scrollView.addView(contentLayout);

        tvContent1 = new TextView(getActivity());
        //tvContent1.text = style
        tvContent1.setTextSize(18F);
        tvContent1.setMovementMethod(LinkMovementMethod.getInstance());
        contentLayout.addView(tvContent1);

        LinearLayout bottomLayout = new LinearLayout(getActivity());
        bottomLayout.setOrientation(LinearLayout.HORIZONTAL);
        RelativeLayout.LayoutParams bottomParams = new RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                DisplayUtil.dp2px(getActivity(), 70)
        );
        bottomParams.addRule(ALIGN_PARENT_BOTTOM);
        bottomLayout.setLayoutParams(bottomParams);
        bottomLayout.setGravity(CENTER);

        tvCancel = new TextView(getActivity());
        tvCancel.setTextSize(18F);
        //tvCancel.text = "Cancel"
        tvCancel.setTextColor(Color.BLACK);
        LinearLayout.LayoutParams cancelParams = new  LinearLayout.LayoutParams(
                0,
                DisplayUtil.dp2px(getActivity(), 50), 1F
        );
        cancelParams.leftMargin = DisplayUtil.dp2px(getActivity(), 20);
        cancelParams.rightMargin = DisplayUtil.dp2px(getActivity(), 20);
        tvCancel.setLayoutParams(cancelParams);
        tvCancel.setGravity(Gravity.CENTER);
        if (Contants.mFREContext != null) {
            tvCancel.setBackground(CustomStyle.getGradientDrawable(Contants.mFREContext));
        } else {
            tvCancel.setBackground(CustomStyle.getGradientDrawable(getActivity()));
        }


        tvOption = new TextView(getActivity());
        tvOption.setTextSize(18F);
        tvOption.setTextColor(Color.BLACK);
        LinearLayout.LayoutParams confirmParams = new  LinearLayout.LayoutParams(
                0,
                DisplayUtil.dp2px(getActivity(), 50), 1F
        );
        tvOption.setLayoutParams(confirmParams);
        tvOption.setGravity(Gravity.CENTER);
        if (Contants.mFREContext != null) {
            tvOption.setBackground(CustomStyle.getGradientDrawable(Contants.mFREContext));
        } else {
            tvOption.setBackground(CustomStyle.getGradientDrawable(getActivity()));
        }

        tvConfirm = new TextView(getActivity());
        tvConfirm.setTextSize(18F);
        //tvConfirm.text = "Confirm"
        tvConfirm.setTextColor(Color.BLACK);
        LinearLayout.LayoutParams more = new  LinearLayout.LayoutParams(
                0,
                DisplayUtil.dp2px(getActivity(), 50), 1F
        );
        more.leftMargin = DisplayUtil.dp2px(getActivity(), 20);
        more.rightMargin = DisplayUtil.dp2px(getActivity(), 20);
        tvConfirm.setLayoutParams(more);
        tvConfirm.setGravity(Gravity.CENTER);
        if (Contants.mFREContext != null) {
            tvConfirm.setBackground(CustomStyle.getGradientDrawable(Contants.mFREContext));
        } else {
            tvConfirm.setBackground(CustomStyle.getGradientDrawable(getActivity()));
        }

        bottomLayout.addView(tvCancel);
        bottomLayout.addView(tvOption);
        bottomLayout.addView(tvConfirm);

        privateLayout.setLayoutParams(rootParams);
        privateLayout.addView(bottomLayout);

        tvCancel.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                parentActivity.mSession.setAgreeFlag(true);
                //Contants.agreeFlag = true;
                parentActivity.cancelConsent();
                getActivity().finish();
            }
        });
        tvConfirm.setOnClickListener(new View.OnClickListener(){

            @Override
            public void onClick(View view) {
                parentActivity.mSession.setAgreeFlag(true);
                //Contants.agreeFlag = true;
                parentActivity.submitFormData();

                if (Contants.mFREContext != null) {
                    getActivity().finish();
                }

                if (parentActivity.mSession.getPermissionFlag()) {
                    getActivity().finish();
                }
            }


        });
        tvOption.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                getActivity().getSupportFragmentManager().beginTransaction()
                        .addToBackStack("option")
                        .replace(Contants.HOME_CONTAINER_ID, new OptionFragment())
                        .commit();
            }
        });

    }

    private void showResponse(final String msg){
        parentActivity.runOnUiThread(new Runnable() {
            @Override
            public void run() {

                Toast.makeText(getActivity(), msg, Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void customActionBarView() {

        actionBarLayout = new  LinearLayout(getActivity());
        RelativeLayout.LayoutParams rootParams = new  RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
        );

        actionBarLayout.setLayoutParams(rootParams);
        actionBarLayout.setGravity(Gravity.CENTER);

        TextView tvTitle = new TextView(getActivity());
        tvTitle.setTextSize(22F);
//        tvTitle.text = "Consent"
        tvTitle.setText("test");
//        tvTitle.setText(parentActivity.info.value.terms_page_title);
        tvTitle.setTextColor(Color.WHITE);
        LinearLayout.LayoutParams titleParams = new  LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
        );
        titleParams.rightMargin = DisplayUtil.dp2px(getActivity(), 50);
        tvTitle.setLayoutParams(titleParams);
        tvTitle.setGravity(Gravity.CENTER);

        actionBarLayout.addView(tvTitle);
    }

    private void setData() {

//        tvTitle.setText(parentActivity.info.value.terms_page_title);
//        tvCancel.setText(parentActivity.info.value.cancel_button_text);
//        tvOption.setText(parentActivity.info.value.option_button_text);
//        tvConfirm.setText(parentActivity.info.value.confirm_button_text);


          tvTitle.setText(parentActivity.mSession.getTermsTitle());
          tvCancel.setText(parentActivity.mSession.getCancelButton());
          tvOption.setText(parentActivity.mSession.getOptionButton());
          tvConfirm.setText(parentActivity.mSession.getConfirmButton());

        tvContent1.setText(Html.fromHtml(parentActivity.mSession.getTermsContent()));
        //String content = "<p>By entering your email and clicking confirm, you consent to the collection of the use of your data to our trusted partners and us. Our trusted partners whom we share the information with may include storage, analytic providers, agencies, platforms, data providers, and research development. The purpose of sharing the data allows our third parties for the following (a) Data Customization: to custom data with demographics, behavioral, contextual or other information for personalized targeted advertisement (b) Measurement: measure key point indicators to evaluate marketing performance (c) Analytics: Identify and analyze behavioral data and patterns, and/or make more-informed business decisions and verify or disprove scientific models, theories and hypotheses (d) Modeling: To pinpoint key shared attributions for look alike audiences (e) Research and Development: allowing parties to process information to create and/or enhance the quality of products (f) Data Management Platform: to create better audiences to target specific users to increase performance When you confirm, you not only grant your consent, you acknowledge you are of 16 years of age and older. Please note, if you choose to click cancel, no information will be collected from you. To learn more about the terms in its entirety, please click <a href=\\\"https://www.motrixi.com/index.php/privacy-policy-2/\\\" target=\\\"_blank\\\" rel=\\\"noopener\\\">here</a>.</p>\\n<p>We thank you for installing our app and helping us improve the user experience by clicking 'Confirm'.</p>";
        //tvContent1.setText(Html.fromHtml(content));
        //tvContent1.setText(Html.fromHtml(parentActivity.info.value.terms_content));
        //tvContent1.setMovementMethod(LinkMovementMethod.getInstance());
        CharSequence str = tvContent1.getText();
        if (str instanceof Spannable) {
            int end = str.length();
            Spannable sp = (android.text.Spannable) tvContent1.getText();
            URLSpan[] urls=sp.getSpans(0, end, URLSpan.class);  //find a tag of the text
            SpannableStringBuilder style=new SpannableStringBuilder(str);
            style.clearSpans(); //should clear old spans
            for(final URLSpan url : urls){

                ClickableSpan clickableSpan = new ClickableSpan() {
                    @Override
                    public void onClick(@NonNull View view) {
                        getActivity().getSupportFragmentManager().beginTransaction()
                                .addToBackStack("show_more")
                                .replace(Contants.HOME_CONTAINER_ID, ShowMoreFragment.newInstance(url.getURL()))
                                .commit();
                    }
                };
                //设置样式其中参数what是具体样式的实现对象，start则是该样式开始的位置，end对应的是样式结束的位置，
                // 参数 flags，定义在Spannable中的常量
                style.setSpan(clickableSpan ,sp.getSpanStart(url),sp.getSpanEnd(url),Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
            }
            tvContent1.setText(style);
        }
    }
}

File: app/src/main/java/com/motrixi/datacollection/fragment/PrivacyStatementFragment.kt
package com.motrixi.datacollection.fragment

import android.annotation.SuppressLint
import android.content.Intent
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.support.v4.app.Fragment
import android.text.*
import android.text.method.LinkMovementMethod
import android.text.style.ClickableSpan
import android.text.style.ForegroundColorSpan
import android.text.style.URLSpan
import android.util.Log
import android.view.Gravity
import android.view.Gravity.CENTER
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import android.widget.RelativeLayout.ALIGN_PARENT_BOTTOM
import android.widget.RelativeLayout.ALIGN_PARENT_RIGHT
import com.motrixi.datacollection.DataCollectionActivity
import com.motrixi.datacollection.R
import com.motrixi.datacollection.adapter.LanguageAdapter
import com.motrixi.datacollection.content.Contants
import com.motrixi.datacollection.content.Session
import com.motrixi.datacollection.network.models.LanguageInfo
import com.motrixi.datacollection.service.MotrixiService
import com.motrixi.datacollection.utils.CustomStyle
import com.motrixi.datacollection.utils.DisplayUtil


// TODO: Rename parameter arguments, choose names that match
// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
private const val ARG_PARAM1 = "param1"
private const val ARG_PARAM2 = "param2"

/**
 * A simple [Fragment] subclass.
 * Use the [PrivacyStatementFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class PrivacyStatementFragment : Fragment(), View.OnClickListener {
    // TODO: Rename and change types of parameters
    private var param1: String? = null
    private var param2: String? = null

    private var rootView: View? = null
    private var parentActivity: DataCollectionActivity? = null

    lateinit var privateLayout: RelativeLayout
    private var actionBarLayout: RelativeLayout? = null
    private var mSession: Session? = null
    lateinit var tvTitle: TextView
    lateinit var tvContent1: TextView
    lateinit var tvCancel: TextView
    lateinit var tvOption: TextView
    lateinit var tvConfirm: TextView

    lateinit var contentLayout: LinearLayout
    lateinit var bottomLayout: LinearLayout
    lateinit var tvLanguage: TextView
    lateinit var listView: ListView


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            param1 = it.getString(ARG_PARAM1)
            param2 = it.getString(ARG_PARAM2)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
//        rootView = inflater.inflate(R.layout.fragment_privacy_statement, container, false)
//        ViewCompat.setElevation(rootView!!, 50f)

        initLayout()
        rootView = privateLayout

        return rootView
    }

    @SuppressLint("ResourceType", "NewApi")
    private fun initLayout() {
        privateLayout = RelativeLayout(activity)
        val rootParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )

        var topLayout = RelativeLayout(activity)
        val topParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            DisplayUtil.dp2px(activity!!, 50)
        )
        topLayout.id = Contants.PRIVACY_TOP_ID
        topLayout.layoutParams = topParams
        topLayout.setBackgroundColor(Color.rgb(0, 150, 182))
        privateLayout.addView(topLayout)

        var ivBack = ImageView(activity)
        val imageParams = RelativeLayout.LayoutParams(
            DisplayUtil.dp2px(activity!!, 25),
            DisplayUtil.dp2px(activity!!, 25)
        )
        imageParams.leftMargin = DisplayUtil.dp2px(activity!!, 20)
        imageParams.addRule(RelativeLayout.CENTER_VERTICAL)
        //ivBack.setImageResource(R.drawable.ic_back)
        ivBack.layoutParams = imageParams
        //topLayout.addView(ivBack)

        tvTitle = TextView(activity)
        val titleParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.WRAP_CONTENT,
            ViewGroup.LayoutParams.WRAP_CONTENT
        )
        titleParams.addRule(RelativeLayout.CENTER_IN_PARENT)
        //tvTitle.text = "Consent"
        tvTitle.textSize = 20F
//        tvTitle.setTextColor(activity!!.resources.getColor(R.color.black))
        tvTitle.setTextColor(Color.BLACK)
        tvTitle.layoutParams = titleParams

        tvLanguage = TextView(activity)
        tvLanguage.textSize = 17F
        tvLanguage.text = "Language"
        tvLanguage.setTextColor(Color.BLACK)
        val languageParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.WRAP_CONTENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )
        languageParams.addRule(ALIGN_PARENT_RIGHT)
        languageParams.rightMargin = DisplayUtil.dp2px(activity!!, 10)
        tvLanguage.layoutParams = languageParams
        tvLanguage.gravity = Gravity.CENTER

        topLayout.addView(tvTitle)
        topLayout.addView(tvLanguage)


        var scrollView: ScrollView = ScrollView(activity!!)
        val scrollParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )
        scrollParams.bottomMargin = DisplayUtil.dp2px(activity!!, 70)
        scrollParams.addRule(RelativeLayout.BELOW, topLayout.id)
        scrollView.layoutParams  =scrollParams
        privateLayout.addView(scrollView)

        contentLayout = LinearLayout(activity)
        contentLayout.orientation = LinearLayout.VERTICAL
        val contentParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )
        contentParams.setMargins(DisplayUtil.dp2px(activity!!, 15),DisplayUtil.dp2px(activity!!, 15),DisplayUtil.dp2px(activity!!, 15),DisplayUtil.dp2px(activity!!, 15))
        contentLayout.layoutParams = contentParams
        scrollView.addView(contentLayout)

        val style = SpannableStringBuilder()
        style.append(Contants.PRIVATE_STATEMENT_1)
        //设置部分文字点击事件
        val clickableSpan: ClickableSpan = object : ClickableSpan() {
            override fun onClick(widget: View) {
                activity!!.supportFragmentManager.beginTransaction()
                    .addToBackStack("show_more")
                    .replace(Contants.HOME_CONTAINER_ID, ShowMoreFragment())
                    .commit()
            }
        }
        style.setSpan(clickableSpan, Contants.PRIVATE_STATEMENT_1.length - 5,
            Contants.PRIVATE_STATEMENT_1.length, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)

//        val colorSpan = ForegroundColorSpan(Color.parseColor(R.color.app_color.toString()))
        val colorSpan = ForegroundColorSpan(Color.rgb(0, 150, 182))
        style.setSpan(colorSpan, Contants.PRIVATE_STATEMENT_1.length - 5,
            Contants.PRIVATE_STATEMENT_1.length, Spannable.SPAN_EXCLUSIVE_INCLUSIVE)


        tvContent1 = TextView(activity)
        //tvContent1.text = style
        tvContent1.textSize = 18F
        tvContent1.movementMethod = LinkMovementMethod.getInstance()

        val tvContent2 = TextView(activity)
        tvContent2.textSize = 18F
        tvContent2.text = Contants.PRIVATE_STATEMENT_2

        contentLayout.addView(tvContent1)
        //contentLayout.addView(tvContent2)

        bottomLayout = LinearLayout(activity)
        bottomLayout.orientation = LinearLayout.HORIZONTAL
        val bottomParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            DisplayUtil.dp2px(activity!!, 70)
        )
        bottomParams.addRule(ALIGN_PARENT_BOTTOM)
        bottomLayout.layoutParams = bottomParams
        bottomLayout.gravity = CENTER

        tvCancel = TextView(activity)
        tvCancel.textSize = 18F
        //tvCancel.text = "Cancel"
        tvCancel.setTextColor(Color.BLACK)
//        tvCancel.setTextColor(Color.rgb(0, 150, 182))  //#0096B6
//        tvCancel.setTextColor(Color.WHITE)
        val cancelParams = LinearLayout.LayoutParams(
            0,
            DisplayUtil.dp2px(activity!!, 50), 1F
        )
        cancelParams.leftMargin = DisplayUtil.dp2px(activity!!, 20)
        cancelParams.rightMargin = DisplayUtil.dp2px(activity!!, 20)
        tvCancel.layoutParams = cancelParams
//        tvCancel.background = activity!!.resources.getDrawable(R.drawable.constant_button_bg)
//        tvCancel.setBackgroundColor(Color.rgb(0, 150, 182))  //#0096B6
        tvCancel.gravity = Gravity.CENTER
        tvCancel.background = CustomStyle.getGradientDrawable(activity!!)

        tvOption = TextView(activity)
        tvOption.textSize = 18F
        //tvOption.text = "Options"
        tvOption.setTextColor(Color.BLACK)
//        tvOption.setTextColor(Color.rgb(0, 150, 182))  //#0096B6
//        tvOption.setTextColor(Color.WHITE)
        val confirmParams = LinearLayout.LayoutParams(
            0,
            DisplayUtil.dp2px(activity!!, 50), 1F
        )
        tvOption.layoutParams = confirmParams
        tvOption.gravity = Gravity.CENTER
        tvOption.background = CustomStyle.getGradientDrawable(activity!!)
//        tvOption.background = activity!!.resources.getDrawable(R.drawable.constant_button_bg)
//        tvOption.setBackgroundColor(Color.rgb(0, 150, 182))  //#0096B6

        tvConfirm = TextView(activity)
        tvConfirm.textSize = 18F
        //tvConfirm.text = "Confirm"
        tvConfirm.setTextColor(Color.BLACK)
//        tvConfirm.setTextColor(Color.rgb(0, 150, 182))  //#0096B6
//        tvConfirm.setTextColor(Color.WHITE)
        val more = LinearLayout.LayoutParams(
            0,
            DisplayUtil.dp2px(activity!!, 50), 1F
        )
        more.leftMargin = DisplayUtil.dp2px(activity!!, 20)
        more.rightMargin = DisplayUtil.dp2px(activity!!, 20)
        tvConfirm.layoutParams = more
        tvConfirm.gravity = Gravity.CENTER
//        tvConfirm.background = activity!!.resources.getDrawable(R.drawable.constant_button_bg)
        tvConfirm.background = CustomStyle.getGradientDrawable(activity!!)
//        tvConfirm.setBackgroundColor(Color.rgb(0, 150, 182))  //#0096B6

        //set style
//        val strokeWidth = DisplayUtil.dp2px(activity!!, 3) // 3dp 边框宽度
//        val roundRadius = DisplayUtil.dp2px(activity!!, 15) // 8dp 圆角半径
//        val strokeColor = Color.parseColor("#2E3135") //边框颜色
//        val fillColor = Color.parseColor("#0096B6") //内部填充颜色
//        val gd = GradientDrawable() //创建drawable
//        gd.setColor(fillColor)
//        gd.cornerRadius = roundRadius.toFloat()
//        gd.setStroke(strokeWidth, strokeColor)
//        tvConfirm.background = gd

        bottomLayout.addView(tvCancel)
        bottomLayout.addView(tvOption)
        bottomLayout.addView(tvConfirm)

        privateLayout!!.layoutParams = rootParams
        privateLayout.addView(bottomLayout)

        listView = ListView(activity)
        val listParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )
        listParams.addRule(RelativeLayout.BELOW, topLayout.id)
        listView.layoutParams = listParams
        listView.visibility = View.GONE
        privateLayout.addView(listView)

        ivBack.setOnClickListener(object : View.OnClickListener{
            override fun onClick(p0: View?) {

                activity!!.finish()
            }
        })

        tvCancel.setOnClickListener(object : View.OnClickListener{
            override fun onClick(p0: View?) {
                mSession!!.agreeFlag = true
                Log.e("stop service", "stop service")
                var startService: Intent = Intent(activity, MotrixiService::class.java)
                activity!!.stopService(startService)
                mSession!!.isCollecting = false

                parentActivity!!.rejectCollect()
                activity!!.finish()
            }
        })
        tvConfirm.setOnClickListener(object : View.OnClickListener{
            override fun onClick(p0: View?) {
                //parentActivity!!.initPermission()
                mSession!!.agreeFlag = true
                var formValue = ""
                if (!mSession!!.isCollecting) {
                    Log.e("start service", "start service")
                    var startService: Intent = Intent(activity, MotrixiService::class.java)
                    if (Build.VERSION.SDK_INT >= 26) {
                        activity!!.startForegroundService(startService)
                    } else {
                        activity!!.stopService(startService)
                    }

                }
                mSession!!.isCollecting = true


                if (parentActivity!!.optionArray.size > 0) {

                    for (index in parentActivity!!.selectedOptionList.indices) {
                        if (index == parentActivity!!.selectedOptionList.size -1) {
                            formValue = formValue + parentActivity!!.selectedOptionList[index]
                        } else {
                            formValue = formValue + parentActivity!!.selectedOptionList[index] + "|"
                        }
                    }

                    parentActivity!!.submitConsentFormData(formValue)
                }
            }
        })
        tvOption.setOnClickListener(object : View.OnClickListener{
            override fun onClick(p0: View?) {
                activity!!.supportFragmentManager.beginTransaction()
                    .addToBackStack("option")
                    .replace(Contants.HOME_CONTAINER_ID, OptionFragment())
                    .commit()
            }
        })

        tvLanguage.setOnClickListener(object : View.OnClickListener{
            override fun onClick(p0: View?) {

                if (parentActivity!!.lanList != null && parentActivity!!.lanList.size > 0) {
                    setListViewData()
                }
            }
        })

    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        parentActivity = activity as DataCollectionActivity?
        mSession = Session(activity!!)
        //initView()

        customActionBarView()
        //parentActivity!!.actionBar!!.customView = actionBarLayout
        //parentActivity!!.actionBar!!.setDisplayShowCustomEnabled(true)

        Log.d("fragment", mSession!!.consentDataInfo.toString())
        setData()

    }


    private fun customActionBarView() {

        actionBarLayout = RelativeLayout(activity)
        val rootParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )

        actionBarLayout!!.layoutParams = rootParams
        actionBarLayout!!.gravity = Gravity.CENTER

        val tvTitle = TextView(activity)
        tvTitle.textSize = 22F
//        tvTitle.text = "Consent"
        tvTitle.text = parentActivity!!.info!!.value!!.terms_page_title
        tvTitle.setTextColor(Color.WHITE)
        val titleParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.WRAP_CONTENT,
            ViewGroup.LayoutParams.WRAP_CONTENT
        )
        titleParams.rightMargin = DisplayUtil.dp2px(activity!!, 50)
        tvTitle.layoutParams = titleParams
//        tvCancel.setBackgroundColor(Color.rgb(0, 150, 182))  //#0096B6
        tvTitle.gravity = Gravity.CENTER

        val tvLanguage = TextView(activity)
        tvLanguage.textSize = 17F
        tvLanguage.text = ""
        tvLanguage.setTextColor(Color.BLACK)
        val languageParams = RelativeLayout.LayoutParams(
            ViewGroup.LayoutParams.WRAP_CONTENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )
        languageParams.addRule(ALIGN_PARENT_RIGHT)
        tvLanguage.layoutParams = languageParams
        tvLanguage.gravity = Gravity.CENTER

        actionBarLayout!!.addView(tvTitle)
        actionBarLayout!!.addView(tvLanguage)

        tvLanguage.setOnClickListener(object : View.OnClickListener{
            override fun onClick(p0: View?) {

                if (parentActivity!!.lanList != null && parentActivity!!.lanList.size > 0) {
                    setListViewData()
                }
            }
        })
    }

    private fun setData() {

        tvTitle.text = parentActivity!!.info!!.value!!.terms_page_title
        tvCancel.text = parentActivity!!.info!!.value!!.cancel_button_text
        tvOption.text = parentActivity!!.info!!.value!!.option_button_text
        tvConfirm.text = parentActivity!!.info!!.value!!.confirm_button_text
        if (!TextUtils.isEmpty(parentActivity!!.info!!.value!!.language_button_text)) {
            tvLanguage.text = parentActivity!!.info!!.value!!.language_button_text
        } else {
            tvLanguage.text = "Language"
        }

        tvContent1.text = Html.fromHtml(parentActivity!!.info!!.value!!.terms_content)
        val str: CharSequence = tvContent1.text
        if (str is Spannable) {
            val end = str.length
            val sp = tvContent1.text as Spannable
            val urls = sp.getSpans(0, end, URLSpan::class.java) //find a tag of the text
            val style = SpannableStringBuilder(str)
            style.clearSpans() //should clear old spans
            for (url in urls) {
                //set click event
                val clickableSpan: ClickableSpan = object : ClickableSpan() {
                    override fun onClick(widget: View) {
                        activity!!.supportFragmentManager.beginTransaction()
                            .addToBackStack("show_more")
                            .replace(Contants.HOME_CONTAINER_ID, ShowMoreFragment.newInstance(url.url.toString()))
                            .commit()
                    }
                }
                style.setSpan(clickableSpan, sp.getSpanStart(url), sp.getSpanEnd(url), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
            }
            tvContent1.text = style
        }
    }

    private fun setListViewData() {
        val lanList: ArrayList<LanguageInfo> = ArrayList()
        lanList.add(LanguageInfo("jan", "Japanese"))
        lanList.add(LanguageInfo("zh", "Chinese"))
        lanList.add(LanguageInfo("en", "English"))

        contentLayout.visibility = View.GONE
        bottomLayout.visibility = View.GONE
        listView.visibility = View.VISIBLE

        listView.adapter = LanguageAdapter(activity, 0, parentActivity!!.lanList)
        listView.onItemClickListener = object : AdapterView.OnItemClickListener{
            override fun onItemClick(p0: AdapterView<*>?, p1: View?, position: Int, p3: Long) {

                var code = parentActivity!!.lanList[position].code.toString()
                Log.e("code", code)
                contentLayout.visibility = View.VISIBLE
                bottomLayout.visibility = View.VISIBLE
                listView.visibility = View.GONE

                parentActivity!!.getConsentDataList(activity!!, code, 1)
            }
        }
    }

    @SuppressLint("ResourceAsColor")
    private fun initView() {

        val style = SpannableStringBuilder()
        style.append(getString(R.string.fragment_private_statement))
        //设置部分文字点击事件
        val clickableSpan: ClickableSpan = object : ClickableSpan() {
            override fun onClick(widget: View) {
                activity!!.supportFragmentManager.beginTransaction()
                    .addToBackStack("show_more")
                    .replace(Contants.HOME_CONTAINER_ID, ShowMoreFragment())
                    .commit()
            }
        }
        style.setSpan(clickableSpan, getString(R.string.fragment_private_statement).length - 5,
            getString(R.string.fragment_private_statement).length, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)

//        val colorSpan = ForegroundColorSpan(Color.parseColor(R.color.app_color.toString()))
        val colorSpan = ForegroundColorSpan(activity!!.resources.getColor(R.color.app_color))
        style.setSpan(colorSpan, getString(R.string.fragment_private_statement).length - 5,
            getString(R.string.fragment_private_statement).length, Spannable.SPAN_EXCLUSIVE_INCLUSIVE)

        /*tv_content.text = style
        tv_content.movementMethod = LinkMovementMethod.getInstance();


        tv_cancel.setOnClickListener(this)
        tv_confirm_all.setOnClickListener(this)
        tv_option.setOnClickListener(this)
        iv_back.setOnClickListener(this)*/
    }

    override fun onClick(view: View?) {
        /*when (view!!.id) {
            R.id.tv_cancel -> {
                activity!!.finish()
            }
            R.id.iv_back -> {
                activity!!.finish()
            }
            R.id.tv_confirm_all -> {

                parentActivity!!.initPermission()

                //activity!!.finish()

            }
            R.id.tv_option -> {
                activity!!.supportFragmentManager.beginTransaction()
                    .addToBackStack("option")
                    .replace(Contants.HOME_CONTAINER_ID, OptionFragment())
                    .commit()
            }
            else -> {
            }
        }*/
    }




    companion object {

        @JvmStatic
        fun newInstance(param1: String, param2: String) =
            PrivacyStatementFragment().apply {
                arguments = Bundle().apply {
                    putString(ARG_PARAM1, param1)
                    putString(ARG_PARAM2, param2)
                }
            }
    }


}